core/src/main/java/org/apache/cxf/common/spi/GeneratedClassClassLoader.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.common.spi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import org.apache.cxf.Bus;
+import org.apache.cxf.common.classloader.ClassLoaderUtils;
+import org.apache.cxf.common.logging.LogUtils;
+import org.apache.cxf.common.util.StringUtils;
+
+public class GeneratedClassClassLoader {
+    private static final Logger LOG = LogUtils.getL7dLogger(ClassLoaderProxyService.class);
+    protected final Bus bus;
+
+    public GeneratedClassClassLoader(Bus bus) {
+        this.bus = bus;
+    }
+    protected Class<?> findClass(String className, Class<?> callingClass) {
+        ClassLoader cl = getClassLoader();
+        try {
+            return cl.loadClass(className);
+        } catch (ClassNotFoundException e) {
+            //ignore and try with other class loader
+        }
+        try {
+            return ClassLoaderUtils.loadClass(className, callingClass);
+        } catch (ClassNotFoundException e) {
+            LOG.fine("Failed to load class :" + e.toString());
+        }
+        return null;
+    }
+    public TypeHelperClassLoader getClassLoader() {
+        TypeHelperClassLoader loader = bus.getExtension(TypeHelperClassLoader.class);
+        if (loader == null) {
+            loader = bus.getExtension(TypeHelperClassLoader.class);
+            if (loader == null) {
+                ClassLoader parent = bus.getExtension(ClassLoader.class);
+                if (parent == null) {
+                    parent = Thread.currentThread().getContextClassLoader();
+                }
+                loader = new TypeHelperClassLoader(parent);
+                bus.setExtension(loader, TypeHelperClassLoader.class);
+            }
+        }
+        return loader;
+    }
+
+    public void restore(ClassGeneratorCapture capture) {
```

Comment:
I don't think you need this method, is it supposed to be used outside of tests?

Reply:
hmm outside of test, I will proceed like that:
- make my own capture
- write in .class file the bytes
- How to inject the class files generated and available in app to the ClassLoader used in GeneratedClassClassLoader.java?
sorry if my question seems dumb but I am a bit lost....

Reply:
Outside of a test you have written the .class in target/classes (maven/gradle) or in classloader singleton of quarkus (= it is int the future runtime classloader, no need to do anything special except have a stable name to reuse it and load it, xxhash64 of all parameters can help there, you can check the openwebbeans impl which has this option). The main trick is to ensure CXF uses a SPI to load the class. Generally the "write"/capture and "read"/load is the same SPI to ensure both sides match. One "already written" impl will throw an UnsupportedOperationException in write but it is a detail. In OWB the read/write methods are merged as a "defineAndLoad" ([1]) one but it is not required. The defineAndLoad deep implementation (actual classloader load) is either the one we currently have with a map in a classloader or a more aggressive one using Unsafe ([2]).

[1] https://github.com/apache/openwebbeans/blob/master/webbeans-impl/src/main/java/org/apache/webbeans/service/ClassLoaderProxyService.java
[2] https://github.com/apache/openwebbeans/blob/master/webbeans-impl/src/main/java/org/apache/webbeans/proxy/Unsafe.java#L132

Reply:
hmm so it mean that I expose Map<String, byte[]> proxies = new HashMap<>(); as they do ?
I do not like to expose byte. I prefer to inject class loader of Map<String, Class<?>> proxies;
because load a byte to class is not really needed at runtime.  no ?

Reply:
You dont have to expose the map, you can provide a getter from the classloader (because it is what these classes do at the end) and it is convenient for libs like yours which will then just dump everything from there but it is not required. Your lib can also extend it and dump it on the fly. The getter does not cost anything and is a simple option but forcing a subclass is not that complicated too. Also note that the runtime mode does NOT use the map of bytes, it is the capture mode, the runtime mode just loadClass directly.

Reply:
I have removed restore response. I guess it is right now ;-)